worker/apicaller/connect.go

```diff
@@ -83,12 +93,18 @@ func connectFallback(
 	conn api.Connection, didFallback bool, err error,
 ) {
 
+	infoCopy := *info
+	if len(infoCopy.Addrs) > 1 {
+		src := rand.NewSource(time.Now().UnixNano())
```

Comment:
the Rand should really live longer than a single connection call, but I guess it's not really a big deal